### PR TITLE
use apiextensions in e2e test

### PR DESCRIPTION
**Pull request by fgimenez**
Uses an interface that unifies functionality from the common clientset and from our api extensions.

### CODE REVIEW

File: integration/client.go [hunk 1]
```diff
@@ -3,14 +3,41 @@
...
 	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/giantswarm/e2e-harness/pkg/harness"
 )
 
-func getK8sClient() (kubernetes.Interface, error) {
+type k8sClient interface {
+	CoreV1() corev1.CoreV1Interface
+	ExtensionsV1beta1() extensionsv1beta1.ExtensionsV1beta1Interface
+	CoreV1alpha1() corev1alpha1.CoreV1alpha1Interface
```
Comment (kopiczko) [inline integration/client.go]: What will happen when we release `CoreV1`?

**Comment by fgimenez** [comment]
Ok, I've updated it to use a client's struct, pls take another look. IMO with this approach we lose the help of the type system (we can pass nil or uninitialized members, that's why we see so many checks on our `new` logic), the signature of the methods is much less expressive this way, but of course I can be wrong, will keep trying to convince you guys anyway! :)

File: integration/client.go (hunk 1 above)
Comment (fgimenez) [reply integration/client.go]: I would iterate, applying the same approach again. At some point our `CoreV1` would differ from upstream's, there I would define the interface and an implementer type that decides which implementation to use for each of the requested methods.

Comment (fgimenez) [reply integration/client.go]: This way we can use a single client for interacting with k8s, which is good for the legibility and maintainability of code (the client code depend on who implements a resource IMO, maybe that can change in the future).

**Comment by kopiczko** [comment]
You can also pass nil or uninitialized members to a function call. Is there any difference?

**Comment by fgimenez** [comment]
Sorry, I was ranting about something else in this case :D not passing the receiver as parameter but struct with embedded dependencies as parameters.

**Pull request merged**